src/shogun/multiclass/KDTreeKNNsolver.cpp

```diff
@@ -65,7 +65,7 @@ int32_t* CKDTREEKNNSolver::classify_objects_k(CDistance* distance, const int32_t
 		for (index_t j=0; j<m_k; j++)
 		{
 			train_lab[j] = m_train_labels[ NN(j,i) ];
-			dists[j] = distance->distance(i, NN(j,i));
+			dists[j] = distance->distance(NN(j,i), i);
```

Comment:
It actually will get an out boundary error[1] if we don't switch it
 [1]**In file /Users/mikeling/shogun/src/shogun/distance/Distance.cpp line 188: idx_a (0) must be in [0,111] and idx_b (103) must be in [0,36]
unknown file: Failure**